lib/trino-parquet/src/main/java/io/trino/parquet/predicate/PredicateUtils.java

```diff
@@ -84,9 +85,9 @@ public static boolean isStatisticsOverflow(Type type, ParquetIntegerStatistics p
         if (type == BIGINT) {
             return false;
         }
-        if (type instanceof DecimalType && ((DecimalType) type).getScale() == 0) {
+        if (type instanceof DecimalType) {
             DecimalType decimalType = (DecimalType) type;
-            if (!decimalType.isShort()) {
+            if (decimalType.getScale() > 0 || !decimalType.isShort()) {
```

Comment:
This changed because this method can be called when a short decimal is used with a scale (for example the DECIMAL(12,2) test in TestParquetPageSkipping. Without this change, that test fails. It can be simplified further too by removing the  `decimalType.getScale() > 0` check above.